src/controllers/signup.js

```diff
@@ -0,0 +1,18 @@
+import { createUser } from './user'
+
+export const signup = async (parent, signupData) => {
+  const authResult = {
+    user: null,
+    jwt: null,
+    authError: null
+  }
+
+  try {
+    authResult.user = await createUser(parent, signupData)
+    authResult.jwt = await authResult.user.generateNewJWT()
+  } catch (authError) {
+    authResult.authError = authError
+  }
+
+  return authResult
+}
```

Comment:
I know that the guide on the training repo tells you to add an `authError` field, but it's not necessary. If you let the error pass-through, Apollo will catch it and return it in a special `error` GraphQL field.
```suggestion
export const signup = async (parent, signupData) => {
  const user = await createUser(parent, signupData)
  const jwt = await user.generateNewJWT()

  return {  user, jwt }
}
```